okapi-core/src/test/java/org/folio/okapi/OkapiPerformance.java

```diff
@@ -55,53 +56,11 @@ public void setUp(TestContext context) {
   @After
   public void tearDown(TestContext context) {
     async = context.async();
-    td(context);
-  }
-
-  public void td(TestContext context) {
-    if (locationAuth != null) {
-      httpClient.delete(port, "localhost", locationAuth, context.asyncAssertSuccess(response -> {
-            context.assertEquals(204, response.statusCode());
-            response.endHandler(x -> {
-              locationAuth = null;
-              td(context);
-            });
-          }));
-      return;
-    }
-    if (locationSample != null) {
-      httpClient.delete(port, "localhost", locationSample, context.asyncAssertSuccess(response -> {
-        context.assertEquals(204, response.statusCode());
-        response.endHandler(x -> {
-          locationSample = null;
-          td(context);
-        });
-      }));
-      return;
-    }
-    if (locationSample2 != null) {
-      httpClient.delete(port, "localhost", locationSample2, context.asyncAssertSuccess(response -> {
-        context.assertEquals(204, response.statusCode());
-        response.endHandler(x -> {
-          locationSample2 = null;
-          td(context);
-        });
-      }));
-      return;
-    }
-    if (locationSample3 != null) {
-      httpClient.delete(port, "localhost", locationSample3, context.asyncAssertSuccess(response -> {
-        context.assertEquals(204, response.statusCode());
-        response.endHandler(x -> {
-          locationSample3 = null;
-          td(context);
-        });
-      }));
-      return;
-    }
-    vertx.close(x -> {
-      async.complete();
-    });
+    httpClient.request(HttpMethod.DELETE, port, "localhost", "/_/discovery/modules")
+        .onComplete(context.asyncAssertSuccess(res -> {
+          vertx.close(x -> context.asyncAssertSuccess());
```

Comment:
Remove `x ->` otherwise this will be called after the test has finished and will therefore not check whether vertx close has failed.

Reply:
good catch